test/test_utils.py

```diff
@@ -338,3 +339,101 @@ def test_expired(self):
           }
         """)
         self.assertEqual(isExpired, True)
+
+
+class TestFromEnv(TestCase):
+
+    def clear_env(self):
+        for v in 'ROOT_URL', 'CLIENT_ID', 'ACCESS_TOKEN', 'CERTIFICATE':
+            v = 'TASKCLUSTER_' + v
+            if v in os.environ:
+                del os.environ[v]
+
+    @mock.patch.dict(os.environ)
+    def test_empty(self):
+        self.clear_env()
+        self.assertEqual(subject.fromEnvVars(), {})
+
+    @mock.patch.dict(os.environ)
```

Comment:
According to the docs, a context manager is available, and patch.dict supports passing in the mocked values directly, which might be easier to understand the intent here than the decorator and os.environ assignments.  It also allows passing in a string `'os.environ'` which would even make importing `os` in this module unneeded.  This is not blocking landing.

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict

Reply:
Yes, but the `values` option does not allow *unsetting* variables, and I don't want to use `clear=True` since the code might require other env vars like HOME or TMPDIR or whatever.  If the code under test starts importing os at runtime, we can change the tests at that time.  So, I'll leave it as-is.